bluefin-internal/src/Bluefin/Internal/Algae.hs

```diff
@@ -0,0 +1,50 @@
+{-# LANGUAGE
+  BangPatterns,
+  GADTs,
+  RankNTypes,
+  ScopedTypeVariables,
+  StandaloneKindSignatures,
+  TypeOperators #-}
+{-# OPTIONS_HADDOCK not-home #-}
+
+-- | = Algebraic effects and named handlers
+module Bluefin.Internal.Algae
+  ( AEffect
+  , HandlerBody
+  , Handler
+  , handle
+  , call
+  ) where
+
+import Data.Kind (Type)
+import Bluefin.Internal (Eff, Effects, type (:&), type (:>))
+import Bluefin.Internal.DelCont
+
+-- | Algebraic effect.
+type AEffect = Type -> Type
+
+-- | Interpretation of an algebraic effect @f@: a function to handle the operations of @f@.
+type HandlerBody :: AEffect -> Effects -> Type -> Type
+type HandlerBody f ss a = (forall x. f x -> (x -> Eff ss a) -> Eff ss a)
+
+-- | Handler to call operations of the effect @f@.
+type Handler :: AEffect -> Effects -> Type
+data Handler f s where
+  Handler :: !(PromptTag ss a s) -> HandlerBody f ss a -> Handler f s
+
+-- | Handle operations of @f@.
+--
+-- === Warning for exception-like effects
+--
+-- If the handler might not call the continuation (like for an exception effect), and
+-- if the handled computation manages resources (e.g., opening files, transactions)
+-- prefer 'handle'' to trigger resource clean up with cancellable continuations.
```

Comment:
Is this referring to something called `handle'`? I don't see what it is ...

Reply:
Right it should be `Bluefin.Algae.Cancellable.handle`